[android] required changes for unified headers & clang

Rechi opened the pull request:
<!--- Provide a general summary of your change in the Title above -->

## Description
Allows building with ndk15 (and newer) and clang.
<!--- Describe your change in detail -->

## Motivation and Context
needed for the switch to ndk16 and clang 
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here -->

## How Has This Been Tested?
compiled for android arm with ndk 14 and ndk 16
<!--- Please describe in detail how you tested your change -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc -->

<!--## Screenshots (if appropriate):-->

## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Improvement (non-breaking change which improves existing functionality)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [x] All new and existing tests passed

koying commented:
Mmm... Any specific reason to create a subset of https://github.com/xbmc/xbmc/pull/12634 ?
As I said, unified header is mostly useless w/o libc++ because of unfixed bugs in gnutls

Rechi commented:
Just to keep PRs focused to one thing.

koying committed:
CLEANUP: [emu_msvcrt] remove non-emulated cases

koying committed:
ADD: [droid] Support NDK 15b+ (unified headers)

samba: ndk 15c has ifaddrs.h but some functions are only supported with sdk 24 or higher. Therefore ignore ifaddrs.h from ndk and still use samba internal logic.

Rechi committed:
[fix] missing --with-ndk to --with-ndk-(api|path) renames at e6c79d3bee

Rechi committed:
[depends] bump libandroidjni (clang build fixes)

koying committed:
FIX: [droid;doc] --ndk-dir no more supported

koying committed:
FIX: [amc] properly unsigned array

Rechi committed:
[cmake][android] clang doesn't support mvectorize-with-neon-quad argument

koying committed:
FIX: [droid] main.cpp is not proper main source

Rechi committed:
[depends] libffi: fix arm assembler syntax

MartijnKaijser commented:
Jenkins build and merge
